commands.go

```diff
@@ -15,49 +15,50 @@ import (
 )
 
 type cmd struct {
-	name     string
-	run      func(line string, u *user)
-	argsInfo string
-	info     string
+	name      string
+	run       func(line string, u *user)
+	argsInfo  string
+	info      string
+	printable bool
 }
 
 var (
 	allcmds = make([]cmd, 30)
 	cmds    = []cmd{
-		{"=<user>", dmCMD, "<msg>", "DM <user> with <msg>"}, // won't actually run, here just to show in docs
-		{"users", usersCMD, "", "List users"},
-		{"color", colorCMD, "<color>", "Change your name's color"},
-		{"exit", exitCMD, "", "Leave the chat"},
-		{"help", helpCMD, "", "Show help"},
-		{"emojis", emojisCMD, "", "See a list of emojis"},
-		{"bell", bellCMD, "on|off|all", "ANSI bell on pings (on), never (off) or for every message (all)"},
-		{"clear", clearCMD, "", "Clear the screen"},
-		{"hang", hangCMD, "<char|word>", "Play hangman"}, // won't actually run, here just to show in docs
-		{"tic", ticCMD, "<cell num>", "Play tic tac toe!"},
-		{"cd", cdCMD, "#room|user", "Join #room, DM user or run cd to see a list"}, // won't actually run, here just to show in docs
-		{"tz", tzCMD, "<zone> [24h]", "Set your IANA timezone (like tz Asia/Dubai) and optionally set 24h"},
-		{"nick", nickCMD, "<name>", "Change your username"},
-		{"theme", themeCMD, "<theme>|list", "Change the syntax highlighting theme"},
-		{"rest", commandsRestCMD, "", "Uncommon commands list"}}
+		{"=<user>", dmCMD, "<msg>", "DM <user> with <msg>", false}, // won't actually run, here just to show in docs
```

Comment:
How would you implement that as there is already multiples groups of slices (`cmds`, `cmdRest`, `secretCMDs, and the lone `cmds` command)? In all those groups there can be commands that need echoing and commands that don't.

One of the way to fix that would be to create a command to boolean map that tells which commands to echo but this would make the code way less maintainable.

Alternatively, there could be two slices for each group of commands. This would be easier to maintain but harder to read.

Reply:
That's a really good point, I'll have to think about this.